python/paddle/fluid/tests/unittests/mkldnn/test_prelu_mkldnn_op.py

```diff
@@ -0,0 +1,149 @@
+        for i in range(x.shape[1]):
+            result[:, i] = np.where(x[:, i] > 0, x[:, i],
+                                    x[:, i] * weight[0, i])
+    elif mode == "element":
+        result = np.where(x[:] > 0, x[:], x[:] * weight)
+
+    return result
+
+
+class TestPReluModeChannelOneDNNOp(OpTest):
+    def init_attrs(self):
+        self.mode = "element"
+        self.alpha = np.random.random((1, 4, 5, 5)).astype("float32")
+
+    def set_dtype_attr(self):
+        pass
+
+    def set_inputs(self):
+        self.inputs = {'X': self.x, 'Alpha': self.alpha}
+
+    def setUp(self):
+        self.op_type = "prelu"
+        self.x = np.random.random((2, 4, 5, 5)).astype("float32") + 1
+        self.init_attrs()
+        self.set_inputs()
+        self.attrs = {'mode': self.mode, 'use_mkldnn': True}
+        self.set_dtype_attr()
+
+        self.outputs = {'Out': ref_prelu(self.x, self.alpha, self.mode)}
+
+    def test_check_output(self):
+        self.check_output()
+
+    def test_check_grad(self):
+        self.check_grad(['X', 'Alpha'], 'Out')
+
+
+class TestPReluModeAllOneDNNOp(TestPReluModeChannelOneDNNOp):
+    def init_attrs(self):
+        self.mode = "all"
+        self.alpha = np.random.random((1, 1, 1, 1)).astype("float32")
+
+    # Skip 'Alpha' input check because in mode = 'all' it has to be a single 1D value, so checking if it has at least 100 values will cause an error
+    def test_check_grad(self):
+        self.check_grad(['X'], 'Out')
+
+
+class TestPReluModeElementOneDNNOp(TestPReluModeChannelOneDNNOp):
+    def init_attrs(self):
+        self.mode = "element"
+        self.alpha = np.random.random((1, 4, 5, 5)).astype("float32")
+
+
+class TestPReluModeChannel3DOneDNNOp(TestPReluModeChannelOneDNNOp):
+    def init_attrs(self):
+        self.mode = "channel"
+        self.x = np.random.random((1, 100, 1)).astype("float32")
+        self.alpha = np.random.random((1, 100, 1)).astype("float32")
+
+
+#   BF16 TESTS
+def create_bf16_test_class(parent):
+    class TestPReluBF16OneDNNOp(parent):
+        def set_inputs(self, ):
+            self.inputs = {
+                'X': convert_float_to_uint16(self.x),
+                'Alpha': convert_float_to_uint16(self.alpha)
+            }
+
+        def set_dtype_attr(self):
+            self.attrs['mkldnn_data_type'] = "bfloat16"
+
+        def calculate_grads(self):
+            dout = self.outputs['Out']
+            self.dx = self.x.copy()
+            self.dalpha = self.alpha.copy()
+
+            if self.mode == "all":
+                self.dx = np.where(self.x > 0, dout, dout * self.alpha[0])
+            elif self.mode == "channel":
+                for i in range(self.x.shape[1]):
+                    self.dx[:, i] = np.where(self.x[:, i] > 0, dout[:, i],
+                                             dout[:, i] * self.alpha[0, i])
+            elif self.mode == "element":
+                self.dx = np.where(self.x[:] > 0, dout[:], dout[:] * self.alpha)
```

Comment:
Isn't it conceptually the same code as for `ref_prelu` ?

Reply:
It's similiar, but arguments of numpy functions are different. I think that merging them together won't result in clear code